crates/rspack_plugin_javascript/src/parser_plugin/import_meta_plugin.rs

```diff
@@ -79,11 +79,15 @@ impl JavascriptParserPlugin for ImportMetaPlugin {
     if for_name == expr_name::IMPORT_META_VERSION {
       Some(eval::evaluate_to_number(5_f64, start, end))
     } else if for_name == expr_name::IMPORT_META_URL {
-      Some(eval::evaluate_to_string(
-        self.import_meta_url(parser),
-        start,
-        end,
-      ))
+      if parser.compiler_options.output.environment.supports_document() {
```

Comment:
The script must be an ES module to use `import.meta.url`. We could not use `supports_document()` to determine this.